src/mlpack/methods/ann/activation_functions/logistic_function.hpp

```diff
@@ -58,7 +58,10 @@ class LogisticFunction
   template<typename InputVecType, typename OutputVecType>
   static void Fn(const InputVecType& x, OutputVecType& y)
   {
-    y = (1.0 / (1 + arma::exp(-x)));
+    y.set_size(arma::size(x));
+
+    for (size_t i = 0; i < x.n_elem; i++)
+      y(i) = Fn(x(i));
```

Comment:
I think allocating fixed shapes and initializing the matrix/vector makes the difference to much extent.  Can you also check by initializing matrices/vectors with zeros? Also, consider ```++i``` in place of ```i++```, as ```++i``` is almost always more efficient than ```i++```. See [this](http://www.cs.technion.ac.il/users/yechiel/c++-faq/increment-pre-post-speed.html)

Reply:
Hi @mrityunjay-tripathi, the difference occurs because matrix multiplication is more heavier compared to simple for loop, changing to i++ or ++I shouldn't result in any significant changes.

Reply:
I will try it just in case.